### PR TITLE
Test Fixes for Date compare on Where tests.

### PR DESCRIPTION
**Pull request by StevePy**
Update for date-based tests to address format differences and cases where the tests can fail due to -0 days + x seconds.

### CODE REVIEW

**Comment by StevePy** [comment]
Additionally, I'd recently come across an issue on StackOverflow that led me to an interesting edit on Linq.Dynamic that doesn't look to have been incorporated into the source. I'm not sure if it just wasn't put forward or it was rejected for some reason... Adding support for a static return type on Select. The article in question is https://stackoverflow.com/questions/1465700/system-linq-dynamic-select-new-into-a-listt-or-any-other-enumerabl with the solution put forward by dahlbyk.  It adds support for a Select<TReturn> so that users can do something like .Select("new ViewModel(Name = name, ...)") using .Select<ViewModel>("new (Name = name, ...)")

I thought there may be a valid reason this hasn't been incorporated, but I thought I'd put the idea forward to you before considering a pull request. From what I saw all of the existing unit tests pass, so I'd just want to take a stab at a new test to cover this off before creating a pull request.

File: Src/System.Linq.Dynamic.Tests/EntitiesTests.cs [hunk 1]
```diff
@@ -189,10 +189,10 @@ public void Entities_Where_ByDateTime_AsTypeAccess()
             var expected = _context.Posts.Where(x => x.PostDate > utcNow).ToArray();
 
             //Act
-            var test = _context.Posts.Where("PostDate > DateTime(\"" + utcNow + "\")").ToDynamicArray();
+            var test = _context.Posts.Where("PostDate > DateTime(\"" + utcNow.ToString("u") + "\")").ToDynamicArray();
 
             //Assert
-            CollectionAssert.AreEqual(expected, test);
+            CollectionAssert.AreEqual(expected, test, string.Format("expected: {0} found: {1}", expected.Length, test.Length));
```
Comment (ghost) [inline Src/System.Linq.Dynamic.Tests/EntitiesTests.cs:195]: If I am not mistaken the CollectionAssert automatically writes the error message regarding count mismatch. Why do we need the message here?

**Comment by ghost** [comment]
I personally simply never had such a use case but I understand its value because even making a second select<dynamic> with property read is a significant performance penalty.
I will review the solution proposed on stackoverflow in details and let you know whether it can be incorporated as-is or we will need to modify it first.

File: Src/System.Linq.Dynamic.Tests/EntitiesTests.cs (hunk 1 above)
Comment (StevePy) [reply Src/System.Linq.Dynamic.Tests/EntitiesTests.cs:195]: The message that it produces just states that the collection counts didn't match... It doesn't really reveal whether the expected collection was off, or the results were. I was still a bit baffled why those two tests failed. They passed when ran individually but failed when ran by TestDriven.Net as part of the suite. I was getting expected = 1, found: 0 when it failed. So Where(x=> x.date > utcNow) was returning a row, where the .Where ("Date > " + utcNow) was not.  First thing that came to mind was possibly M/D/Y vs D/M/Y as I'm based in Australia but possibly ms conversion on values that were just > today. (-0 days, + 1 sec??)  Had those tests ever failed when run /w the entire set?

**Comment by StevePy** [comment]
Sounds good, let me know what you decide. The feature is implemented on my fork master branch if you want to have a look at the code. When I had a look the only thing that looked a bit off was that it appeared that it would record the type in all cases when that type was parsed, not just in the case of the .Select<Type>() . Not sure if that could pose any unintended side-effects.

File: Src/System.Linq.Dynamic.Tests/EntitiesTests.cs (hunk 1 above)
Comment (ghost) [reply Src/System.Linq.Dynamic.Tests/EntitiesTests.cs:195]: Actually these tests were never failed. I was tried to run the separately and in batch and still same result. I will continue to dig into.

Comment (ghost) [reply Src/System.Linq.Dynamic.Tests/EntitiesTests.cs:195]: I wasn't able to confirm the need to fix DateTime tests (I wasn't able to reproduce it nether on my machine nether in Azure Dev Ops).

**Comment by ghost** [comment]
Support for static or strongly typed result is implemented as part of #7. The new package version is available on NuGet (https://www.nuget.org/packages/AK.System.Linq.Dynamic/1.1.32.2)

**Pull request closed without merging**